src/content.ts

```diff
@@ -464,6 +468,16 @@ function cancelSponsorSchedule(): void {
 function startSponsorSchedule(includeIntersectingSegments: boolean = false, currentTime?: number): void {
     cancelSponsorSchedule();
 
+    // Don't skip if advert playing and reset last checked time
+    if (isAdPlaying) {
+        // Reset lastCheckVideoTime
+        lastCheckVideoTime = -1;
+        lastCheckTime = 0;
+
+        lastVideoTime = video.currentTime;
```

Comment:
Shouldn't this be `lastVideoTime = null`?

Actually, it seems that variable is unused entirely and can probably just be removed.

Reply:
I thought the variable was used since it was also in the pause event listener, but it turns out its unused so its removed now